crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -2007,6 +2126,37 @@ async fn update_payment_method_status_and_ntid<F: Clone>(
     Ok(())
 }
 
+fn update_connector_mandate_details_for_the_flow<F: Clone>(
+    connector_mandate_id: Option<String>,
+    mandate_metadata: Option<serde_json::Value>,
+    payment_data: &mut PaymentData<F>,
+) -> RouterResult<()> {
+    let connector_mandate_reference_id = Some(ConnectorMandateReferenceId {
+        connector_mandate_id: connector_mandate_id.clone(),
+        payment_method_id: None,
+        update_history: None,
+        mandate_metadata: mandate_metadata.clone(),
+    });
+
+    payment_data.payment_attempt.connector_mandate_detail = connector_mandate_reference_id
+        .as_ref()
+        .map(Encode::encode_to_value)
+        .transpose()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Error encoding the connector mandate details")
+        .map_err(|err| logger::error!(err=?err))
+        .ok()
+        .flatten();
+
+    payment_data.set_mandate_id(api_models::payments::MandateIds {
```

Comment:
we should update this in the payment_method instead of payment_data right?

Reply:
No , every payment_attempt would have its own connector_mandate_id, and the payment_method would only have the id if and only if its successful payment